dev-lang/php/patches/php-7.4.16.patchset

```diff
@@ -0,0 +1,796 @@
+From 786d243c65c614a2294f3520898c34482679d740 Mon Sep 17 00:00:00 2001
+From: <pulkomandy@pulkomandy.tk>
+Date: Thu, 4 Mar 2021 23:40:05 +0000
+Subject: Build fixes
+
+
+diff --git a/configure.ac b/configure.ac
+index e1adda7..546621f 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -393,9 +393,9 @@ dnl Some systems (OpenServer 5) dislike -lsocket -lnsl, so we try to avoid -lnsl
+ dnl checks, if we already have the functions which are usually in libnsl. Also,
+ dnl uClibc will bark at linking with glibc's libnsl.
+ 
+-PHP_CHECK_FUNC(socket, socket)
+-PHP_CHECK_FUNC(socketpair, socket)
+-PHP_CHECK_FUNC(htonl, socket)
++PHP_CHECK_FUNC(socket, network)
++PHP_CHECK_FUNC(socketpair, network)
++PHP_CHECK_FUNC(htonl, network)
+ PHP_CHECK_FUNC(gethostname, nsl)
+ PHP_CHECK_FUNC(gethostbyaddr, nsl)
+ PHP_CHECK_FUNC(dlopen, dl)
+diff --git a/ext/fileinfo/libmagic/ascmagic.c b/ext/fileinfo/libmagic/ascmagic.c
+index 0bc0cce..bbd925d 100644
+--- a/ext/fileinfo/libmagic/ascmagic.c
++++ b/ext/fileinfo/libmagic/ascmagic.c
+@@ -51,7 +51,7 @@ FILE_RCSID("@(#)$File: ascmagic.c,v 1.104 2019/05/07 02:27:11 christos Exp $")
+ #define ISSPC(x) ((x) == ' ' || (x) == '\t' || (x) == '\r' || (x) == '\n' \
+ 		  || (x) == 0x85 || (x) == '\f')
+ 
+-private unsigned char *encode_utf8(unsigned char *, size_t, unichar *, size_t);
++private unsigned char *encode_utf8(unsigned char *, size_t, php_unichar *, size_t);
```

Comment:
this part of the patch could be removed after beta3 is released (the incorrect unichar definition in Haiku headers has been removed)